### PR TITLE
feat(svelte-query-devtools): Svelte Adapter for new Devtools

### PR DESCRIPTION
**Pull request by lachlancollins**
Implements #5347 for Svelte 🚀

### CODE REVIEW

**Comment by TkDodo** [comment]
wohoo 🚀 you guys are on fire 🔥

**Comment by ardeora** [comment]
This is amazing! Let me know if I can help in anyway possible 🙏

**Comment by lachlancollins** [comment]
@ardeora everything works fine in dev mode in SvelteKit, then as soon as I build for prod I get `ReferenceError: window is not defined`. I've tried several things to only get it to run in dev on the browser with no luck yet. Would you have any suggestions?

**Comment by ardeora** [comment]
> @ardeora everything works fine in dev mode in SvelteKit, then as soon as I build for prod I get `ReferenceError: window is not defined`. I've tried several things to only get it to run in dev on the browser with no luck yet. Would you have any suggestions?

I think this is expected right? We shouldn't be including the devtools in the prod build! If you see this part you'll see that we conditionally remove the devtools panel on build! https://github.com/TanStack/query/blob/alpha/packages/react-query-devtools/src/index.ts#L5-L10 

Is there a way to do this in Svelte where it treeshakes the devtools in prod environment?

**Comment by lachlancollins** [comment]
> Is there a way to do this in Svelte where it treeshakes the devtools in prod environment?

That's what I'm trying to do, my first two methods didn't work - I'll search around for a diff solution!

**Comment by ardeora** [comment]
> > Is there a way to do this in Svelte where it treeshakes the devtools in prod environment?
> 
> That's what I'm trying to do, my first two methods didn't work - I'll search around for a diff solution!

Oh yeah I'm not super familiar with Svelte but I'll try to find solutions that might make this work. I'm curious what are the two methods you already tried?

**Comment by lachlancollins** [comment]
> Oh yeah I'm not super familiar with Svelte but I'll try to find solutions that might make this work. I'm curious what are the two methods you already tried?

On mobile so apologies for formatting:

1. Tried process.env.NODE_ENV like react devtools but didn't expect it to work
2. Tried esm-env (which is also used by SvelteKit) to recognise when running on browser/server/dev/prod
3. Tried using {#if dev && browser}<SvelteQueryDevtools />{/if} within the svelte-basic example

**Comment by lachlancollins** [comment]
`esm-env` definitely seems like the best way to get it working (https://kit.svelte.dev/docs/packaging#best-practices). Maybe I need a dynamic import - even importing the component without rendering it was triggering the error.

**Comment by lachlancollins** [comment]
@ardeora @TkDodo this is ready to go - is there anything I need to set for it to be published automatically?

File: packages/svelte-query-devtools/src/Devtools.svelte [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      const { TanstackQueryDevtools } = await import('@tanstack/query-devtools')
+      const devtools = new TanstackQueryDevtools({
+        client,
+        queryFlavor: 'Svelte Query',
+        version: '5',
+        onlineManager,
+        buttonPosition,
+        position,
+        initialIsOpen,
+        errorTypes,
```
Comment (ardeora) [inline packages/svelte-query-devtools/src/Devtools.svelte:31]: Super everything is looking great!! I was wondering if the props are reactive. Say if there is a button in the parent conatiner that changes buttonPosition from 'top-left' -> 'top-right'. Will the devtools button switch position?

Comment (lachlancollins) [reply packages/svelte-query-devtools/src/Devtools.svelte:33]: They are now!

**Review by ardeora — APPROVED** [review]

**Pull request merged**